epsagon/modules/gunicorn.py

```diff
@@ -0,0 +1,191 @@
+"""
+Gunicorn patcher module.
+"""
+
+from __future__ import absolute_import
+import warnings
+import time
+import uuid
+import traceback
+
+try:
+    from gunicorn.config import (
+        PreRequest, validate_callable, PostRequest, validate_post_request
+    )
+except ImportError:
+    class PreRequest(object):
+        """
+        Dummy PreRequest Class
+        """
+        def __init__(self):
+            pass
+
+    class PostRequest(object):
+        """
+        Dummy Post Request Class
+        """
+        def __init__(self):
+            pass
+
+    def validate_callable(_):
+        """
+        Dummy Validate Callable Function
+        :param _:
+        :return:
+        """
+        return True
+
+    def validate_post_request(_):
+        """
+        Dummy Validate Post Request Function
+        :param _:
+        :return:
+        """
+        return True
+
+import epsagon
+import epsagon.trace
+from epsagon.common import EpsagonWarning
+from epsagon.utils import collect_container_metadata
+
+
+from ..event import BaseEvent
+from ..http_filters import ignore_request
+from ..utils import add_data_if_needed
+
+
+class GunicornRunner(BaseEvent):
+    """
+    Represents Python Gunicorn event runner.
+    """
+
+    ORIGIN = 'runner'
+    RESOURCE_TYPE = 'python_gunicorn'
+    OPERATION = 'request'
+
+    def __init__(self, start_time, request):
+        """
+        Initialize.
+        :param start_time: event's start time (epoch).
+        :param request: the incoming request.
+        """
+        super(GunicornRunner, self).__init__(start_time)
+
+        self.event_id = str(uuid.uuid4())
```

Comment:
Starting this point this looks like a general http based code (getting the hostname, the path and the headers). Can we inherit an HttpEvent instead of BaseEvent? (Extract mutual code for http instrumentations)

Reply:
We can't. The data looks different than a generic HTTP request and response object as HttpEvent expects it to be, it didn't work.